src/utilities.js

```diff
@@ -250,3 +251,92 @@ export function getResults(request) {
     });
   };
 }
+
+export function getTrendingResults(request) {
+  return (searchType, obj) => {
+    const map = {
+      'Daily trends': {
+        path: '/trends/api/dailytrends',
+        extraParams: {
+          ed: convertDateToString(obj.trendDate, false, true),
+          ns: obj.ns,
+        },
+      },
+      'Real time trends': {
+        path: '/trends/api/realtimetrends',
+        extraParams: {
+          fi: 0,
+          fs: 0,
+          ri: 300, // # of trending stories IDs returned
+          rs: 20,
+          sort: 0,
+        },
+      },
+    };
+
+    let options = {
+      method: 'GET',
+      host: 'trends.google.com',
+      path: map[searchType].path,
+      qs: {
+        hl: obj.hl,
+        tz: obj.timezone,
+        geo: obj.geo,
+        cat: obj.category,
+      },
+    };
+
+    if (obj.agent) options.agent = obj.agent;
+
+    Object.assign(options.qs, options.qs, map[searchType].extraParams);
+
+    return request(options)
+    .then((res) => {
+      try {
+        /** JSON.parse will decode unicode */
+        const results = JSON.stringify(JSON.parse(res.slice(5)));
+
+        return results;
+      } catch (e) {
+        /** throws if not valid JSON, so just return unaltered res string */
+        return res;
+      }
+    });
+  };
+}
+
+export function constructTrendingObj(obj, cbFunc) {
+  if (typeof obj === 'function') cbFunc = obj;
+
+  if (!obj || !!obj && typeof obj !== 'object' || Array.isArray(obj)) {
+    obj = new Error('Must supply an object');
+  }
+
+  if (!!cbFunc && typeof cbFunc !== 'function') {
+    obj = new Error('Callback function must be a function');
+  }
+
+  if (!obj.geo) {
+    obj = new Error('Must supply an geographical location (geo)');
+  }
+
+  if (!obj.hl) obj.hl = 'en-US';
```

Comment:
I try to shy away from mutation these days. Would you mind changing this to copy the `obj` and create a new one using the [optional object parameters](https://stackoverflow.com/questions/47892127/succinct-concise-syntax-for-optional-object-keys-in-es6-es7) trick?

Something along the lines of this:
```
const constructedObj = {
  ...obj,
  ...(!('hl' in obj) && { hl: 'en-US' }),
  ...(!('category' in obj) && { category: 0 }),
```

Reply:
Actually, in this case I think it would be better to have a "defaults" object. Let's do something like this:
```
const defaults = {
  hl: 'en-US',
  category: 0,
  timezone: new Date().getTimezoneOffset(),
  trendDate: new Date(),
  ns: 15,
};

const constructedObj = {
  ...defaultsObj,
  ...obj
};
```

Then the defaults will be overwritten if they exist on `obj`

Reply:
Ok, the second example is semi clear and I'll add a comment explaining what this does. Sorry to be a stickler here, but I'm using this for employment reasons, and I can not risk having code that isn't verbose in it's intention.

Reply:
OK i tried this, and a problem occurs when the we need to signal a logical Error to be returned to the user:
`
> var a = new Error('test');
> a.hl = 12;
> a.category= 0;
> a
{ Error: Test
    at repl:1:5
    at Script.runInThisContext (vm.js:96:20)
    at REPLServer.defaultEval (repl.js:329:29)
    at bound (domain.js:396:14)
    at REPLServer.runBound [as eval] (domain.js:409:12)
    at REPLServer.onLine (repl.js:642:10)
    at REPLServer.emit (events.js:187:15)
    at REPLServer.EventEmitter.emit (domain.js:442:20)
    at REPLServer.Interface._onLine (readline.js:290:10)
    at REPLServer.Interface._line (readline.js:638:8) hl: 12, category: 0 }
> b = { fx: 1, cd: 2};
{ fx: 1, cd: 2 }
> test = { ...b, ...a };
{ fx: 1, cd: 2, hl: 12, category: 0 }
> test
{ fx: 1, cd: 2, hl: 12, category: 0 }
'

As you can see here we lose the ERROR object's identity. Node.js treats the Error obj as an object and just merges the properties. Your orginal code has the Error + additional properties; thus retaining the fact that this is an error.
 
`

Reply:
figured a way around it ... moving forward.